Leaking mlflow dependency

```
tests/conftest.py:4: in <module>
    from rikai.spark.sql import init
../rikai/python/rikai/__init__.py:19: in <module>
    from rikai.spark.sql.codegen import mlflow_logger as mlflow
../rikai/python/rikai/spark/sql/codegen/mlflow_logger.py:19: in <module>
    import mlflow
E   ModuleNotFoundError: No module named 'mlflow'
```

--- a/python/rikai/spark/sql/codegen/mlflow_logger.py
+++ b/python/rikai/spark/sql/codegen/mlflow_logger.py
@@ -16,8 +16,6 @@ Rikai SQL ML
 """
 from typing import Any, Optional
 
-import mlflow
-
 CONF_MLFLOW_TRACKING_URI = "rikai.sql.ml.registry.mlflow.tracking_uri"
 CONF_MLFLOW_OUTPUT_SCHEMA = "rikai.output.schema"
 CONF_MLFLOW_SPEC_VERSION = "rikai.spec.version"
@@ -95,6 +93,9 @@ class MlflowLogger:
 
         For more details see `mlflow docs <https://www.mlflow.org/docs/latest/python_api/mlflow.pytorch.html#mlflow.pytorch.log_model>`_.
         """  # noqa E501
+
+        import mlflow
+
         # no need to set the tracking uri here since this is intended to be
         # called inside the training loop within mlflow.start_run
         getattr(mlflow, flavor).log_model(

--- a/python/rikai/spark/sql/codegen/mlflow_logger.py
+++ b/python/rikai/spark/sql/codegen/mlflow_logger.py
@@ -96,12 +96,12 @@ class MlflowLogger:
 
         try:
             import mlflow
-        except ImportError:
+        except ImportError as e:
             raise ImportError(
                 "Couldn't import mlflow. Please make sure to "
                 "`pip install mlflow` explicitly or install "
                 "the correct extras like `pip install rikai[mlflow]`"
-            )
+            ) from e
 
         # no need to set the tracking uri here since this is intended to be
         # called inside the training loop within mlflow.start_run

--- a/python/rikai/spark/sql/codegen/mlflow_logger.py
+++ b/python/rikai/spark/sql/codegen/mlflow_logger.py
@@ -94,7 +94,14 @@ class MlflowLogger:
         For more details see `mlflow docs <https://www.mlflow.org/docs/latest/python_api/mlflow.pytorch.html#mlflow.pytorch.log_model>`_.
         """  # noqa E501
 
-        import mlflow
+        try:
+            import mlflow
+        except ImportError:
+            raise ImportError(
+                "Couldn't import mlflow. Please make sure to "
+                "`pip install mlflow` explicitly or install "
+                "the correct extras like `pip install rikai[mlflow]`"
+            )
 
         # no need to set the tracking uri here since this is intended to be
         # called inside the training loop within mlflow.start_run

